crates/wadm/src/nats.rs

```diff
@@ -40,6 +39,16 @@ impl From<StreamPersistence> for StorageType {
     }
 }
 
+impl From<&str> for StreamPersistence {
+    fn from(persistance: &str) -> Self {
+        match persistance {
+            "file" => StreamPersistence::File,
+            "memory" => StreamPersistence::Memory,
+            _ => StreamPersistence::File,
+        }
+    }
+}
+
 /// Creates a NATS client from the given options
 pub async fn get_client_and_context(
```

Comment:
Should this function be `pub` now that it is in a crate we release? I think this is more of an internal function. Probably should also check that for the other functions in this file